src/main/client/type.c

```diff
@@ -1554,6 +1555,41 @@ static int set_rack_aware_config(as_config *conf, PyObject *config_dict)
         }
         conf->rack_id = (int)rack_id;
     }
+
+    PyObject *rack_ids_pylist = PyDict_GetItemString(config_dict, "rack_ids");
+    if (rack_ids_pylist == NULL) {
+        return INIT_SUCCESS;
+    }
+    Py_INCREF(rack_ids_pylist);
+
+    if (!PyList_Check(rack_ids_pylist)) {
+        return INIT_POLICY_PARAM_ERR;
+    }
+
+    size_t size = PyList_Size(rack_ids_pylist);
+
+    for (size_t i = 0; i < size; i++) {
+        PyObject *rack_id_pyobj = PyList_GetItem(rack_ids_pylist, i);
+        if (rack_id_pyobj == NULL) {
+            // This shouldn't happen, but just return an error if it does
+            return INIT_POLICY_PARAM_ERR;
+        }
+
+        Py_INCREF(rack_id_pyobj);
+        if (PyLong_Check(rack_id_pyobj) == false) {
+            return INIT_POLICY_PARAM_ERR;
+        }
+
+        long rack_id = PyLong_AsLong(rack_id_pyobj);
+        if (rack_id == -1) {
+            // Error occurred
+            return INIT_POLICY_PARAM_ERR;
+        }
+        as_config_add_rack_id(conf, (int)rack_id);
+        Py_DECREF(rack_id_pyobj);
+    }
+    Py_DECREF(rack_ids_pylist);
```

Comment:
I think `Py_INCREF(rack_ids_pylist);` causes reference leaks in the error cases with returns. You could either decref before the returns or use a cleanup goto and do it there.